### PR TITLE
feat: add linea sepolia network and deprecate Linea Goerli network

### PR DESCRIPTION
**Pull request by VGau**
## **Description**

- A new Linea Sepolia network will be released soon. This PR adds support for the Linea Sepolia network in the MetaMask mobile app.
- Deprecate Linea Goerli network

## **Related issues**

Fixes:

## **Manual testing steps**

For Linea Sepolia:
1. Select Linea Sepolia network
2. Send a transaction
3. Check the transaction result in the explorer
4. Make sure the Linea Sepolia network is display in the network settings page

For Linea Goerli deprecation:
1. Select Linea Goerli network
2. Check that Linea Goerli deprecation message is display at the bottom of the screen

### **After**
https://github.com/MetaMask/metamask-mobile/assets/85494462/9e7c4298-fa47-44fb-9fce-a37b519c5936

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/images/image-icons.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ export default {
   BNB,
   AETH,
   AVAX,
-  'LINEA-GOERLI': LINEA_GOERLI,
+  'LINEA-GOERLI': LINEA_TESTNET,
```
Comment (sethkfman) [inline app/images/image-icons.js:24]: Why are you leaving `LINEA-GOERLI` throughout the code? Are there plans in another PR to remove?

File: app/util/networks/index.js [hunk 2]
```diff
@@ -82,7 +83,16 @@ const NetworkList = {
...
     networkType: 'linea-goerli',
-    imageSource: lineaGoerliLogo,
+    imageSource: lineaTestnetLogo,
+  },
+  [LINEA_SEPOLIA]: {
+    name: 'Linea Sepolia Test Network',
+    shortName: 'Linea Sepolia',
+    networkId: 59141,
+    chainId: toHex('59141'),
+    color: '#61dfff',
```
Comment (sethkfman) [inline app/util/networks/index.js:93]: @brianacnguyen @georgeweiler Do you have a suggestion where this color can be pulled from? I know your team is trying to centralize all colors for the rebranding.

File: app/images/image-icons.js (hunk 1 above)
Comment (VGau) [reply app/images/image-icons.js:24]: The Linea Goerli network will be removed in another PR. For now, I added the Linea goerli network to the deprecated networks.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**